src/interpreter.rs

```diff
@@ -760,7 +760,9 @@ impl<'source> Interpreter<'source> {
                 value,
                 collection,
             } => self.eval_some_in(span, key, value, collection, stmts)?,
-            Literal::NotExpr { expr, .. } => matches!(self.eval_expr(expr)?, Value::Bool(false)),
+            Literal::NotExpr { expr, .. } => {
```

Comment:
Add link to the discussion https://github.com/open-policy-agent/opa/issues/1622#issuecomment-520547385 in comments here.